projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/floating/LLVM80BitFloat.java

```diff
@@ -626,4 +642,108 @@ public static LLVM80BitFloat fromString(String stringValue) {
         return fromBytesBigEndian(DatatypeConverter.parseHexBinary(stringValue));
     }
 
+    protected abstract static class LLVM80BitFloatNativeCallNode extends LLVMNode {
+        private final String name;
+
+        @Child private Node nativeExecute = Message.createExecute(2).createNode();
+
+        public LLVM80BitFloatNativeCallNode(String name) {
+            this.name = name;
+        }
+
+        protected TruffleObject createFunction() {
+            LLVMContext context = getContext();
+            NFIContextExtension nfi = context.getContextExtension(NFIContextExtension.class);
+            return nfi.getNativeFunction(context, "@__sulong_fp80_" + name, "(UINT64,UINT64,UINT64):VOID");
```

Comment:
isn't that creating a native function for every callsite?
So, if the application does an `add` at 100 places, we get 100 TruffleObjects (native function objects) - we could share that.

(I just realized that `LLVMNativeFunction` is doing the same thing - I'll fix `LLVMNativeFunction`)